gcloud-java-dns/src/main/java/com/google/gcloud/dns/ChangeRequest.java

```diff
@@ -38,280 +30,163 @@
   }
 
   /**
-   * Returns the status of this {@code ChangeRequest}.
+   * Applies this change request to a zone that it is associated with.
    */
-  public Status status() {
-    return status;
-  }
-
-  com.google.api.services.dns.model.Change toPb() {
-    com.google.api.services.dns.model.Change pb =
-        new com.google.api.services.dns.model.Change();
-    // set id
-    if (id() != null) {
-      pb.setId(id());
-    }
-    // set timestamp
-    if (startTimeMillis() != null) {
-      pb.setStartTime(ISODateTimeFormat.dateTime().withZoneUTC().print(startTimeMillis()));
-    }
-    // set status
-    if (status() != null) {
-      pb.setStatus(status().name().toLowerCase());
-    }
-    // set a list of additions
-    pb.setAdditions(Lists.transform(additions(), RecordSet.TO_PB_FUNCTION));
-    // set a list of deletions
-    pb.setDeletions(Lists.transform(deletions(), RecordSet.TO_PB_FUNCTION));
-    return pb;
+  public ChangeRequest applyTo(Dns.ChangeRequestOption... options) {
+    return dns.applyChangeRequest(zoneName, this, options);
   }
 
-  static ChangeRequest fromPb(com.google.api.services.dns.model.Change pb) {
-    Builder builder = builder();
-    if (pb.getId() != null) {
-      builder.id(pb.getId());
-    }
-    if (pb.getStartTime() != null) {
-      builder.startTimeMillis(DateTime.parse(pb.getStartTime()).getMillis());
-    }
-    if (pb.getStatus() != null) {
-      // we are assuming that status indicated in pb is a lower case version of the enum name
-      builder.status(ChangeRequest.Status.valueOf(pb.getStatus().toUpperCase()));
-    }
-    if (pb.getDeletions() != null) {
-      builder.deletions(Lists.transform(pb.getDeletions(), RecordSet.FROM_PB_FUNCTION));
-    }
-    if (pb.getAdditions() != null) {
-      builder.additions(Lists.transform(pb.getAdditions(), RecordSet.FROM_PB_FUNCTION));
-    }
-    return builder.build();
+  @Override
+  public Builder toBuilder() {
+    return new Builder(this);
   }
 
   @Override
-  public boolean equals(Object other) {
-    return (other instanceof ChangeRequest) && toPb().equals(((ChangeRequest) other).toPb());
+  public boolean equals(Object obj) {
+    return obj instanceof ChangeRequest
+        && Objects.equals(options, ((ChangeRequest) obj).options)
+        && Objects.equals(zoneName, ((ChangeRequest) obj).zoneName)
+        && Objects.equals(toPb(), ((ChangeRequest) obj).toPb());
   }
 
   @Override
   public int hashCode() {
-    return Objects.hash(additions, deletions, id, startTimeMillis, status);
+    return Objects.hash(super.hashCode(), options, zoneName);
   }
 
-  @Override
-  public String toString() {
-    return MoreObjects.toStringHelper(this)
-        .add("additions", additions)
-        .add("deletions", deletions)
-        .add("id", id)
-        .add("startTimeMillis", startTimeMillis)
-        .add("status", status)
-        .toString();
+  private void readObject(ObjectInputStream in) throws IOException, ClassNotFoundException {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Avoid prefixing parameters by in, out or inOut. Uses Javadoc to document this behavior.](https://www.codacy.com/app/mziccard/gcloud-java/file/2311199854/issues/source?bid=3123106&fileBranchId=3198278#l182)